libafl_targets/src/cmplog.h

```diff
@@ -50,12 +51,16 @@ extern uint8_t libafl_cmplog_enabled;
 void __libafl_targets_cmplog_instructions(uintptr_t k, uint8_t shape,
                                           uint64_t arg1, uint64_t arg2);
 
-void __libafl_targets_cmplog_routines(uintptr_t k, uint8_t *ptr1,
-                                      uint8_t *ptr2);
+void __libafl_targets_cmplog_routines(uintptr_t k, const uint8_t *ptr1,
+                                      const uint8_t *ptr2);
+
+void __libafl_targets_cmplog_routines_len(uintptr_t k, const uint8_t *ptr1,
+                                          const uint8_t *ptr2, size_t len);
 
 static inline void __libafl_targets_cmplog(uintptr_t k, uint8_t shape,
                                            uint64_t arg1, uint64_t arg2) {
   if (!libafl_cmplog_enabled) { return; }
+  libafl_cmplog_enabled = false;
```

Comment:
i think this is something turned on/off by the observer

Reply:
Yup! This is to prevent infinite recursion (it happened in a particularly rare case where I instrumented libc)

Reply:
ok makes sense